lib/tools-common.in

```diff
@@ -658,21 +658,24 @@ initialize_devices() {
 
     if [ "$INITIAL_SETUP" = "yes" ]; then
 	if [ "$DATAPART" != "TBD" ]; then
-	    local partdev="/dev/$DATADEV$DATAPART"
-	    local datatmp=$(mktemp -q -d /tmp/datasave.XXXXXX)
-	    if mount $partdev $datatmp >/dev/null 2>&1; then
+	# Save data partition if data tarball doesn't exist
+	if [ ! -f "$DATA_URL" ]; then
```

Comment:
Rather than a separate `if` here, just move the added condition up to the one at line 660, so it's
```
if [ "$DATAPART" != "TBD" -a -z "$DATA_URL" ]; then
```
and note that just checking if `DATA_URL` is the null string is enough to indicate that there's no data tarball.  `-f` isn't the right test there, since it's a URL rather than just a filename.